packages/components/src/client/components/ArtPlayer.ts

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable vue/no-unused-properties */
+import { PropType, defineComponent, h, onBeforeUnmount, onMounted } from "vue";
+import { deepMerge } from "vuepress-shared/client";
+import { useSize } from "../composables/index.js";
+import { autoType, flv, m3u8, mpd } from "../utils/index.js";
+
+import type { VNode } from "vue";
+
+import type {
+  ArtPlayer,
+  ArtPlayerOptions,
+  ComponentsArtPlayerOptions,
+  CustomArtPlayerOptions,
+} from "../../shared/index.js";
+
+declare const ART_PLAYER: ComponentsArtPlayerOptions;
+
+const optionsDefault = {
+  width: "100%",
+  ratio: 16 / 9,
+  src: {
+    fullscreen: true,
+    autoSize: true,
+    setting: true,
+    whitelist: ["*"],
+  },
+};
+
+export default defineComponent({
+  name: "ArtPlayer",
+
+  props: {
+    /**
+     * artplayer src
+     */
+    src: {
+      type: Object as PropType<CustomArtPlayerOptions>,
+      default: optionsDefault.src,
+      required: false,
+    },
+
+    /**
+     * artplayer src.url
+     */
+    url: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.poster
+     */
+    poster: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.title
+     */
+    title: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.muted
+     */
+    muted: {
+      type: Boolean,
+      default: false,
+      required: false,
+    },
+
+    /**
+     *  return artplayer
+     */
+    player: {
+      type: Function,
+      default: undefined,
+      required: false,
+    },
+
+    /**
+     * artplayer danmuku
+     */
+    pluginDanmuKu: {
+      type: Object,
+      default: () => ART_PLAYER.pluginDanmuKu || {},
+      required: false,
+    },
+
+    /**
+     * Component width
+     *
+     * 组件宽度
+     */
+    width: {
+      type: [String, Number],
+      default: ART_PLAYER?.width || optionsDefault.width || undefined,
+      required: false,
+    },
+
+    /**
+     * Component height
+     *
+     * 组件高度
+     */
+    height: {
+      type: [String, Number],
+      default: ART_PLAYER?.height || undefined,
+      required: false,
+    },
+
+    /**
+     * Component width / height ratio
+     *
+     * 组件长宽比
+     */
+    ratio: {
+      type: [String, Number],
+      default: ART_PLAYER?.ratio || optionsDefault.ratio || undefined,
+      required: false,
+    },
```

Comment:
What' this, optionsDefault.radio is always true.

Also yous should create a reactive object for  these 3 props.

Reply:
> What' this, optionsDefault.radio is always true.
> 
> Also yous should create a reactive object for these 3 props.

I think I need to load the default settings on the node side, and then pass in the client, so it will be simple

Reply:
I personally think we do not need a gloabl settings for this. It's not relastic for user's video to all be another radio